src/psyclone/psyir/nodes/routine.py

```diff
@@ -65,27 +63,22 @@ class Routine(Schedule):
     _children_valid_format = "[Statement]*"
     _text_name = "Routine"
 
-    def __init__(self, name, is_program=False, return_type=None, parent=None):
+    def __init__(self, name, is_program=False, parent=None):
```

Comment:
I could do that but then I would need to also add any supplied symbol to the SymbolTable of the Routine we're constructing. That starts to feel like too much to do in a constructor but if you are happy then I will do it.

Reply:
I think the argument is still nice to have to initialize a function Node, but you are right that constructor should only deal with their internal structure, not the associated symbol tables. But is this a problem? Calling the setter already checks if the provided symbol is in the symbol table. What I propose is just a parameter that goes directly to the setter, having it in the right symbol table is something the user has to do before to initialize a proper function, otherwise call the create method. Does this make sense?

Reply:
I can see what you're saying but the user is unable to supply a symbol table to the constructor - one is automatically created by the underlying `Schedule` constructor. Therefore, if they did supply a return symbol then the check that it is in the symbol table would always fail.